### PR TITLE
[1.24] vendor: bump runc to 1.1.6

### PR DESCRIPTION
**Pull request by kolyshkin**
This is a manual backport of PRs #110496 and #117241 to release-1.24 branch.

#### What type of PR is this?

/kind bug

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Fix "dbus: connection closed by user" error after dbus daemon restart
CVE-2023-27561 CVE-2023-25809 CVE-2023-28642: Bump fix runc v1.1.4 -> v1.1.5
Fixed cgroup removal error when using runc binary >= 1.1.6
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve

File: vendor/github.com/seccomp/libseccomp-golang/seccomp_internal.go [hunk 1]
```diff
@@ -27,10 +26,10 @@ import (
...
 #include <seccomp.h>
 
-#if SCMP_VER_MAJOR < 2
-#error Minimum supported version of Libseccomp is v2.2.0
-#elif SCMP_VER_MAJOR == 2 && SCMP_VER_MINOR < 2
-#error Minimum supported version of Libseccomp is v2.2.0
+#if (SCMP_VER_MAJOR < 2) || \
+    (SCMP_VER_MAJOR == 2 && SCMP_VER_MINOR < 3) || \
+    (SCMP_VER_MAJOR == 2 && SCMP_VER_MINOR == 3 && SCMP_VER_MICRO < 1)
+#error This package requires libseccomp >= v2.3.1
```
[the remark this answers is not in the document]
Comment (kolyshkin) [reply vendor/github.com/seccomp/libseccomp-golang/seccomp_internal.go:32]: This is a compile-time, not run-time check (although there's a run-time check as well).

This is coming from https://github.com/seccomp/libseccomp-golang/commit/449387b6d401d2ca8e231ae21e10ff5ef75ac86d, please see commit message for analysis. TL;DR: at this time, no supported distro comes with version of libseccomp < 2.3.1.

[the remark this answers is not in the document]
Comment (BenTheElder) [reply vendor/github.com/seccomp/libseccomp-golang/seccomp_internal.go:32]: I don't think kubernetes releases have advertised this so 1.24 doesn't seem special versus the other backports?

From the link and checking again the situation today (which hasn't changed much), Ubuntu 14.04 which is still supported but only in extended security maintenance mode [until early 2024](https://ubuntu.com/about/release-cycle) doesn't have 2.3.1+, But 16.04+ do, and other major distros do.

**Comment by liggitt** [comment]
/approve
for dependency mechanics

/hold for node ack of minimum libseccomp bump in a patch release (discussion in https://github.com/kubernetes/kubernetes/pull/117892#discussion_r1190462369)

File: vendor/github.com/seccomp/libseccomp-golang/seccomp_internal.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (kolyshkin) [reply vendor/github.com/seccomp/libseccomp-golang/seccomp_internal.go:32]: Ah, sorry, in my previous comment I missed the fact that Ubuntu 14.04 is still supported.

In any case, libseccomp or libseccomp-golang is not used directly by kubernetes, this is merely a runc/libcontainer dependency. It is not used by runc/libcontainer either, unless the `seccomp` build tag is set, and some specific functions are called.

In other words, this dependency bump is just noise. I'm thinking about a possible way to remove this, but this doesn't look easy.

[the remark this answers is not in the document]
Comment (BenTheElder) [reply vendor/github.com/seccomp/libseccomp-golang/seccomp_internal.go:32]: Regarding build tags: It looks like we only set `selinux` (`notest` is a tag used internally to compile out awkward fuzzer support code), though technically a distro could override this in their non-standard build
https://github.com/kubernetes/kubernetes/blob/53636bc7806d6e4fa1e9a4824d04779d0c4cde39/hack/lib/golang.sh#L875

Thanks for the PRs and for looking into how to remove them!

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Comment by BenTheElder** [comment]
I really don't think this PR caused it but a nil pointer panic in tests is concerning https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/117892/pull-kubernetes-e2e-kind-ipv6/1657109581307842560

**Pull request merged**